packages/cli/commands/project/migrateApp.js

```diff
@@ -80,62 +82,65 @@ exports.handler = async options => {
           migrateApp: true,
         });
 
-  const publicApps = await fetchPublicAppOptions(accountId, accountName);
-  const selectedApp = publicApps.find(a => a.id === appId);
-  const appName = selectedApp ? selectedApp.name : 'app';
-  if (!selectedApp) {
-    logger.error(i18n(`${i18nKey}.errors.invalidAppId`, { appId }));
-    process.exit(EXIT_CODES.ERROR);
-  }
-  if (selectedApp.listingInfo) {
-    logger.error(i18n(`${i18nKey}.errors.invalidMarketplaceApp`, { appId }));
-    process.exit(EXIT_CODES.ERROR);
-  }
-
-  const { name, location } = await createProjectPrompt('', options, true);
+  try {
```

Comment:
This is a pretty massive try catch block. It might make our lives easier down the road if you break this up into smaller, more focused blocks. My assumption is that you moved this up here to include the call to `fetchPublicAppMetadata `.

Reply:
That's right--I can definitely break this up!

Reply:
@brandenrodgers, I'm having a little trouble breaking up the try...catch block for two reasons:

1) I need access to the `selectedApp` (ln. 86), so I can display the warning later (ln. 122). 
2) I need access to `projectLocation` (ln. 100) when downloading the zipped project files (ln. 163). 

Any advice on this?

Reply:
If I'm understanding correctly, you should be able to declare the variables outside the scope of the try/catch block (i.e. use `let`)

Reply:
Oof just failed scoping 🤦‍♀️

Reply:
Addressed in https://github.com/HubSpot/hubspot-cli/pull/1111/commits/b556b9c5946602db821e510edb18917e57231b76